Python/bytecodes.c

```diff
@@ -3026,14 +3045,19 @@ dummy_func(
             /* If the eval breaker is set then stay in tier 1.
              * This avoids any potentially infinite loops
              * involving _RESUME_CHECK */
-            if (_Py_atomic_load_uintptr_relaxed(&tstate->eval_breaker) & _PY_EVAL_EVENTS_MASK) {
+            if (IS_JIT_TRACING() || _Py_atomic_load_uintptr_relaxed(&tstate->eval_breaker) & _PY_EVAL_EVENTS_MASK) {
```

Comment:
This doesn't seem right. What is supposed to happen when tracing reaches `ENTER_EXECUTOR`?
It seems unrelated to the eval breaker being set.